tracer/test/Datadog.Trace.Tests/Ci/Ipc/CircularChannelTests.cs

```diff
@@ -0,0 +1,111 @@
+// <copyright file="CircularChannelTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using Datadog.Trace.Ci.Ipc;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Ci.Ipc;
+
+public class CircularChannelTests
+{
+    private const int BufferSize = 8192;
+    private const int HeaderSize = 4;
+    private const int AvailableBufferSize = BufferSize - HeaderSize;
+
+    public static IEnumerable<object[]> GetWriteData()
+    {
+        var random = new Random();
+        for (var i = 311; i < AvailableBufferSize; i += 311)
+        {
+            var bytes = new byte[i];
+            random.NextBytes(bytes);
+            yield return [bytes];
+        }
```

Comment:
I would suggest also adding some specific edge cases:

```suggestion
        }
        
        foreach(var edgeCase in [0, 1, 2, 3, 4, 8190, 8191, 8192, 8193])
        {
            var bytes = new byte[i];
            random.NextBytes(bytes);
            yield return [bytes];
        }
```

Some of those will be rejected of course - we should have explict tests for those!

Reply:
done in [d4f6c55](https://github.com/DataDog/dd-trace-dotnet/pull/5537/commits/d4f6c55e4bac83eb5d81c9b0aa572c574533279b)